core-bundle/src/Controller/Backend/FavoriteController.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller\Backend;
+
+use Contao\BackendUser;
+use Contao\Controller;
+use Contao\CoreBundle\Exception\BadRequestException;
+use Contao\CoreBundle\Util\UrlUtil;
+use Contao\DC_Table;
+use Doctrine\DBAL\Connection;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\AsController;
+use Symfony\Component\Routing\Attribute\Route;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+#[AsController]
+#[Route('%contao.backend.route_prefix%/_favorites', 'contao_backend_favorites', defaults: ['_scope' => 'backend'])]
+class FavoriteController extends AbstractController
+{
+    public function __construct(
+        private readonly Connection $connection,
+        private readonly UrlGeneratorInterface $urlGenerator,
+    ) {
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $user = $this->getUser();
+
+        if (!$user instanceof BackendUser) {
+            return new Response();
+        }
+
+        if ($request->isMethod(Request::METHOD_POST) && 'remove-favorite' === $request->request->get('FORM_SUBMIT')) {
+            $url = UrlUtil::getNormalizePathAndQuery($request->request->get('target_path'));
+            $id = $this->getCurrentId($url, $user);
+
+            if (!$id) {
```

Comment:
```suggestion
            if ($id === null) {
```
Feels more strict to me, feel free to ignore

Reply:
To me, too, but I think that's rector's work.